benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/goal/explorationspace/workload/users/UserValuesYamlProtocol.scala

```diff
@@ -0,0 +1,162 @@
+            case _ if yamlObject.getFields(RangeKey).nonEmpty =>
+              getRangeList(
+                yamlObject.getFields(RangeKey).headOption.map(_.convertTo[List[Int]]).get).get
+
+          },
+          keyString(ValuesKey))
+
+      } yield UserValues(values = values)
+
+    }
+
+    override def write(obj: Try[UserValues]): YamlValue = unsupportedWriteOperation
+  }
+
+  implicit object UserValuesWriteFormat extends YamlFormat[UserValues] {
+
+    override def write(obj: UserValues): YamlValue = YamlObject {
+
+      Map[YamlValue, YamlValue](
+        ValuesKey -> obj.values.toYaml)
+
+    }
+
+    override def read(yaml: YamlValue): UserValues = unsupportedReadOperation
+
+  }
+
+  private def assertValidValues(values: List[Int]): Try[List[Int]] = {
+
+    // check that all values are in increasing order
+    val increasingOrder = values.sliding(2).forall {
+      case List(x, y) => x < y
+    }
+
+    if (!increasingOrder) {
+      Failure(new BenchFlowDeserializationException("values must be increasing", new Throwable))
+    } else if (values.head <= 0) {
+      Failure(new BenchFlowDeserializationException("values must be > 0", new Throwable))
+    } else {
+      Success(values)
+    }
+
+  }
+
+  private def getRangeList(range: List[Int]): Try[List[Int]] = {
+
+    if (range.size != 2) {
+      Failure(new BenchFlowDeserializationException("range must be specified as a List of 2 elements", new Throwable))
+    } else if (range.head >= range.last) {
+      Failure(new BenchFlowDeserializationException("range must be increasing", new Throwable))
+    } else if (range.head <= 0) {
+      Failure(new BenchFlowDeserializationException("range must be > 0", new Throwable))
+    } else {
+      // IF only range then the step is 1
+      Success((range.head to range(1)).toList)
+    }
+
+  }
+
+  private def getStepList(range: List[Int], step: String): Try[List[Int]] = {
+
+    if (range.size != 2) {
+      Failure(new BenchFlowDeserializationException("range must be specified as a List of 2 elements", new Throwable))
+    } else if (range.head >= range.last) {
+      Failure(new BenchFlowDeserializationException("range must be increasing", new Throwable))
+    } else if (range.head <= 0) {
+      Failure(new BenchFlowDeserializationException("range must be > 0", new Throwable))
+    } else {
+
+      val prefix = step.substring(0, 1)
+
+      val triedValue = Try(step.substring(1).toInt)
+
+      triedValue match {
+
+        case Success(value) =>
+          if (value <= 0) {
+            Failure(new BenchFlowDeserializationException("step must be > 0", new Throwable))
+          } else {
+            generateStepList(range, value, prefix)
+          }
+
+        case Failure(ex) => Failure(new BenchFlowDeserializationException("step must contain an integer", ex))
+
+      }
+
+    }
+
+  }
+
+  private def generateStepList(range: List[Int], step: Int, prefix: String): Try[List[Int]] = prefix match {
+    case "+" =>
```

Comment:
All the checks are generic and can go in the superclass, only the actual function is specific but it is always the same given the type and can be called by the supertype.